python/tank/deploy/tank_commands/console_utils.py

```diff
@@ -190,95 +190,16 @@ def _get_configuration_recursive(log, tank_api_instance, new_ver_descriptor, par
-                        # - {self}/path/to/foo.py
-                        # - {config}/path/to/foo.py
-                        # - {tk-framework-perforce_v1.x.x}/path/to/foo.py
-                        #
-                        # for new style hooks, copy the value across into the
-                        # environment. Resolve the {engine_name} token if it
-                        # exists.
-                        if constants.TANK_HOOK_ENGINE_REFERENCE_TOKEN in default_value:
-                            # we have something on the form {self}/path/to/{engine_name}_foo.py
-                            # replace {engine_name} in the default value with the current engine name
-                            # also check that this file actually exist as part of the app.
-                            # if it doesn't, don't populate the value with a default because it 
-                            # will generate a hook-not-found runtime error
-                            if parent_engine_name is None:
-                                # should not happen
-                                raise TankError("Cannot resolve dynamic engine token for "
-                                                "setting %s in %s!" % (param_name, new_ver_descriptor))
-                            
-                            # Before we can check that the hook file exists, ensure that the code exists...
-                            if not new_ver_descriptor.exists_local():
-                                log.info("Hang on, downloading %s..." % new_ver_descriptor)
-                                new_ver_descriptor.download_local()
-                            
-                            # bake out the current engine token
-                            resolved_default_value = default_value.replace(constants.TANK_HOOK_ENGINE_REFERENCE_TOKEN, 
-                                                                           parent_engine_name)
-                            
-                            # ensure that it exists as part of the app
-                            hooks_folder = os.path.join(new_ver_descriptor.get_path(), "hooks")
-                            full_path = resolved_default_value.replace("{self}", hooks_folder)
-                            
-                            if not os.path.exists(full_path):
-                                log.warning("Sorry, no built-in support for the %s engine yet! " 
-                                            "The default hook value '%s' for this hook refers to an engine specific hook, "
-                                            "however there is currently "
-                                            "no hook implementation available for the current engine. "
-                                            "If you want to continue with the install, you have to supply your "
-                                            "own hooks." % (parent_engine_name, default_value))
```

Comment:
Hmm, so I think maybe the rationale here is that this check happens at install time, where you don't refer to any hooks other than `{self}`. The default value in an app manifest could technically be `{config}/foo/bar.py` but it wouldn't make any sense I think because it assumes certain things about the config you are about to install into! The only thing we know for sure exists is `{self}`. I think this is why the code only covers that case. Would have been nice with some comments in the existing code explaining this!

If we move this logic so that it is resolved at runtime, we should validate all syntaxes!